lib/vnet/daemon/client_darwin.go

```diff
@@ -276,7 +295,15 @@ func startByCalling(ctx context.Context, bundlePath string, config Config) error
 		C.StartVnet(&req, &res)
 
 		if !res.ok {
-			errC <- trace.Errorf("could not start VNet daemon: %v (%v)", C.GoString(res.error_description), C.GoString(res.error_domain))
+			errorDomain := C.GoString(res.error_domain)
+			errorCode := int(res.error_code)
+
+			if errorDomain == vnetErrorDomain && errorCode == int(errorCodeAlreadyRunning) {
+				errC <- trace.Wrap(ErrAlreadyRunning)
+				return
+			}
+
+			errC <- trace.Errorf("could not start VNet daemon: %v", C.GoString(res.error_description))
```

Comment:
`res.error_description` comes from `[error description]` which already reports all fields of that object (including the code and the domain too). `description` seems to be an `NSObject` method that `NSError` reimplements. I believe if we were interested in the actual description alone, we're supposed to use [`localizedDescription`](https://developer.apple.com/documentation/foundation/nserror/1414418-localizeddescription?language=objc).

But I think `description` might be better, it also includes `userInfo` dictionary. In case of some esoteric errors, it might include some extra data which we'd otherwise not get.

Reply:
Sounds good.